### PR TITLE
Test herbstclient command line parsing

### PR DESCRIPTION
**Pull request by t-wissmann**
This extends the tests for most command line flags of the herbstclient
command. Also, herbstclient now prints an error message if too few
parameters are passed.

### CODE REVIEW

File: tests/test_herbstclient.py [hunk 1]
```diff
@@ -139,3 +139,83 @@ def test_lastarg_only(hlwm, repeat):
...
+                              universal_newlines=True)
+        assert proc.returncode != 0
+        assert proc.stdout == ''
+        assert proc.stderr == ''
+
+
+def test_quiet_with_hlwm_running(hlwm):
+    # even with --quiet, herbstclient must produce output
+    for flag in ['-q', '--quiet']:
+        cmd = [HC_PATH, flag, '-q', 'echo', 'test']
```
Comment (blarz) [inline tests/test_herbstclient.py:168]: Why is -q in the parameters, shouldn't flag cover this already?

Comment (blarz) [reply tests/test_herbstclient.py:168]: Same for the test above

**Review by blarz — CHANGES_REQUESTED** [review]

File: tests/test_herbstclient.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_herbstclient.py:168]: I've looked at the coverage report and `-q` was not handled. I think the issue was the wrong parameter (it's fixed by #1206). Also, this case ensures that the stderr is really empty (i.e. that there is no other error).

Comment (blarz) [reply tests/test_herbstclient.py:168]: Yes, but I thought the for-loop is there for iterating over -q and --quiet?
Now cmd is called with ['-q', '-q'] and then ['--quiet', 'q']. 

Maybe I'm overlooking or misunderstanding something

Comment (t-wissmann) [reply tests/test_herbstclient.py]: Aah, you're completely right. I understood your question entirely wrong (and thus started looking for other test cases where `--quiet` is mentioned...). I've fixed it, sorry for the confusion :)

Comment (blarz) [reply tests/test_herbstclient.py]: Thanks! But a nice side-effect of uncovering a different testing error :D

**Review by blarz — APPROVED** [review]

**Pull request merged**